src/MLearning.h

```diff
@@ -92,7 +92,7 @@ namespace prlearn {
             node_t() = default;
             node_t(const node_t& other, size_t dimen);
             node_t(node_t&& other) noexcept = default;
-            node_t& operator=(node_t&& other) noexcept = default;
+            node_t& operator=(node_t&& other) = default;
 
```

Comment:
I guess clang's issue is with a combination of `noexcept` and `default`.
Could you supply the implementation?
it's should be as easy as a sequence of `std::swap` on all the fields.

Reply:
alternatively, do `field = std::move(other.field);`